src/python_testing/matter_testing_infrastructure/matter/testing/apps.py

```diff
@@ -143,32 +159,128 @@ def __init__(self, app: str, rpc_server_port: int, storage_dir: str,
 class OTAProviderSubprocess(AppServerSubprocess):
     """Wrapper class for starting an OTA Provider application server in a subprocess."""
 
-    DEFAULT_ADMIN_NODE_ID = 112233
-
     # Prefix for log messages from the OTA provider application.
     PREFIX = b"[OTA-PROVIDER]"
 
-    def __init__(self, app: str, storage_dir: str, discriminator: int,
-                 passcode: int, ota_source: Union[OtaImagePath, ImageListPath],
-                 port: int = 5541, extra_args: list[str] = []):
-        """Initialize the OTA Provider subprocess.
+    def __init__(
+        self,
+        ota_file: str,
+        discriminator: int,
+        passcode: int,
+        secured_device_port: int,
+        queue: str = None,
+        timeout: int = None,
+        override_image_uri: str = None,
+        log_file_path: str = "provider.log",
+        app_path: str = None,
+        kvs_path: str = None,
+    ):
+        """
+        Initialize OTA Provider with hardcoded KVS path, log file, and extra args.
 
         Args:
-            app: Path to the chip-ota-provider-app executable
-            storage_dir: Directory for persistent storage
-            discriminator: Discriminator for commissioning
-            passcode: Passcode for commissioning
-            port: UDP port for secure connections (default: 5541)
-            ota_source: Either OtaImagePath or ImageListPath specifying the OTA image source
-            extra_args: Additional command line arguments
+            ota_file: Path to OTA image file (string or OtaImagePath).
+            discriminator: Commissioning discriminator.
+            passcode: Setup PIN code.
+            secured_device_port: Port for provider process.
+            queue: Optional queue name.
+            timeout: Optional timeout in seconds.
+            override_image_uri: Optional ImageURI override.
```

Comment:
what is this overriding? Is there a default? how is this different than the constructed uri from the ota_file?